File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
"""define a Rectangle"""


class Rectangle:
    """Rectangle class"""

    def __init__(self, width=0, height=0):
        """define for Rectangle"""
        self.width = width
        self.height = height

    def __str__(self):
        """I need the rectangle"""
        if self.__height == 0 or self.__width == 0:
            return ""
        size = "#" * self.__width
        rect = []
        for index in range(self.__height):
            rect.append(size)
        return "\n".join(rect)

    @property
    def width(self):
        """Rectangle width"""
        return self.__width

    @width.setter
    def width(self, value):
        """Rectangle width"""
        if type(value) != int:
            raise TypeError("width must be an integer")
        if value < 0:
            raise ValueError("width must be >= 0")
        self.__width = value

    @property
    def height(self):
        """Rectangle height"""
        return self.__height

    @height.setter
    def height(self, value):
        """Rectangle height"""
        if type(value) != int:
            raise TypeError("height must be an integer")
        if value < 0:
            raise ValueError("height must be >= 0")
        self.__height = value

    def area(self):
        """Rectangle area"""
        return self.__height * self.__width

    def perimeter(self):
        """returns the perimeter of the Rectangle"""
        if self.__height == 0 or self.__width == 0:
            return
        return (self.__height * 2) + (self.__width * 2)

File: 0x13-javascript_objects_scopes_closures/3-rectangle.js
#!/usr/bin/node
class Rectangle {
  constructor (wt, ht) {
    if (wt > 0 && ht > 0) {
      this.width = wt;
      this.height = ht;
    }
  }

  print () {
    let e;
    let k;
    let build = '';
    for (e = 0; e < this.height; e++) {
      if (e > 0) {
        build += '\n';
      }
      for (k = 0; k < this.width; k++) {
        build += 'X';
      }
    }
    console.log(build);
  }
}
module.exports = Rectangle;
